pkg/authentication/scram/scram_enabler.go

```diff
@@ -4,44 +4,33 @@ import (
 	"errors"
 
 	"github.com/hashicorp/go-multierror"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/util/authtypes"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/util/constants"
+
 	"github.com/mongodb/mongodb-kubernetes-operator/pkg/automationconfig"
 	"github.com/mongodb/mongodb-kubernetes-operator/pkg/util/contains"
 )
 
-const (
-	Sha256                                = "SCRAM-SHA-256"
-	Sha1                                  = "MONGODB-CR"
-	AutomationAgentKeyFilePathInContainer = "/var/lib/mongodb-mms-automation/authentication/keyfile"
-	automationAgentWindowsKeyFilePath     = "%SystemDrive%\\MMSAutomation\\versions\\keyfile"
-	AgentName                             = "mms-automation"
-	AgentPasswordKey                      = "password"
-	AgentKeyfileKey                       = "keyfile"
-)
-
-// configureScramInAutomationConfig updates the provided auth struct and fully configures Scram authentication.
-func configureScramInAutomationConfig(auth *automationconfig.Auth, agentPassword, agentKeyFile string, users []automationconfig.MongoDBUser, opts Options) error {
-	if err := validateScramOptions(opts); err != nil {
+// enableAgentAuthentication updates the provided auth struct and configures scram authentication based on the provided
+// values and configuration options.
+func enableAgentAuthentication(auth *automationconfig.Auth, agentPassword, agentKeyFileContents string, opts authtypes.Options) error {
+	if err := validateAgentOptions(opts); err != nil {
 		return err
 	}
-	enableAgentAuthentication(auth, agentPassword, agentKeyFile, users, opts)
-	enableDeploymentMechanisms(auth, opts)
-	return nil
-}
 
-// enableAgentAuthentication updates the provided auth struct and configures scram authentication based on the provided
-// values and configuration options.
-func enableAgentAuthentication(auth *automationconfig.Auth, agentPassword, agentKeyFileContents string, users []automationconfig.MongoDBUser, opts Options) {
 	auth.Disabled = false
 	auth.AuthoritativeSet = opts.AuthoritativeSet
 	auth.KeyFile = opts.KeyFile
 
 	// windows file is specified to pass validation, this will never be used
-	auth.KeyFileWindows = automationAgentWindowsKeyFilePath
+	auth.KeyFileWindows = constants.AutomationAgentWindowsKeyFilePath
 
-	for _, authMode := range opts.AutoAuthMechanisms {
-		if !contains.String(auth.AutoAuthMechanisms, authMode) {
-			auth.AutoAuthMechanisms = append(auth.AutoAuthMechanisms, authMode)
-		}
+	auth.AutoAuthMechanisms = make([]string, 0)
+	if contains.String(opts.AuthMechanisms, constants.Sha256) {
```

Comment:
These checks on `opts.AuthMechanisms` indicate that they could be individual methods for the `AuthMechanisms` type.

Reply:
The `AuthMechanisms` field is just a string array. Do you mean a method for the `Options` type?

Reply:
Yes, or a helper method to not repeat this logic. It is indeed checking whether a string is contained in the list, but semantically you are checking if a specific auth mechanism is enabled.

Reply:
Done!